### PR TITLE
feature/864_repo_refactoring_multi_agent___doc

### PR DESCRIPTION
**Pull request by frbattid**
* Partially implements issue #864 
* No tests are required
* The result of this documentation refactor can seen at http://fiware-cygnus.readthedocs.org/en/feature-864_repo_refactoring_multi_agent___doc/index.html
* Assignee @pcoello25 , but LGTM from @jmcanterafonseca is also required

### CODE REVIEW

File: doc/cygnus-twitter/to_be_done.md [hunk 1]
```diff
@@ -0,0 +1 @@
+#To be done
```
Comment (jmcanterafonseca) [inline doc/cygnus-twitter/to_be_done.md]: 'Coming soon' better

File: doc/contributing/contributing_guidelines.md [hunk 2]
```diff
@@ -188,7 +189,66 @@ Work in progress.
...
+
+The following elements SHOULD be present as well:
+
+* `doc/<agent name>/flume_extensions_catalogue/` → For each class added to the native Apache Flume library, there SHOULD be a document placed here. Markdown MUST be used for the documents within this subfolder.
+* `doc/<agent name>/quick_start_guide.md/` → Simple and ready-to-use commands/actions to be taken in order to quickly test the agent SHOULD be documented here. Markdown MUST be used.
+
+[Top](#top)
+
+###<a name="section9.1"></a>`readthedocs.org` documentation
+The documentation within the `doc/` folder MUST be exposed in the Interned by means of `readthedocs.org` service. In order to achieve this, a `mkdocs.yml` file MUST live in the root of the main repository acting as a hook.
```
Comment (jmcanterafonseca) [inline doc/contributing/contributing_guidelines.md]: MUST be published to readthedocs.org .

File: doc/contributing/contributing_guidelines.md [hunk 3]
```diff
@@ -188,7 +189,66 @@ Work in progress.
 [Top](#top)
 
 ##<a name="section9"></a>Documentation
-Work in progress.
+###<a name="section9.1"></a>Repository documentation
```
Comment (jmcanterafonseca) [inline doc/contributing/contributing_guidelines.md]: Every Cygnus Agent MUST include a documentation folder as per the following rules:

File: doc/contributing/contributing_guidelines.md [hunk 4]
```diff
@@ -188,7 +189,66 @@ Work in progress.
...
+There MUST exist a `doc/` folder at the root of the main repository. Within this folder, there will be a subfolder for each agent type, in addition to a subfolder for `cygnus-common`. For instance:
+
+* `doc/cygnus-ngsi`
+* `doc/cygnus-twitter`
+* `doc/cygnus-common`
+
+All the agent teams MUST provide any documentation aimed to be exposed to the Community within its documentation subfolder.
+
```
Comment (jmcanterafonseca) [inline doc/contributing/contributing_guidelines.md]: I think this sentence is redundant

File: mkdocs.yml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+site_description: Cygnus Documentation
+docs_dir: doc
+site_dir: html
+markdown_extensions: [toc,fenced_code]
+use_directory_urls: false
+theme: readthedocs
+extra_css: ["https://fiware.org/style/fiware_readthedocs.css"]
+pages:
+  - Home: index.md
+  - 'Contributing guidelines': 'contributing/contributing_guidelines.md'
```
Comment (jmcanterafonseca) [inline mkdocs.yml]: perhaps we can link this from index.md, by having a section on index.md and a link to this document from that section, what do you think?

I find it a bit weird to have the contribution guidelines so high in the readthedocs TOC.

**Comment by pcoello25** [comment]
LGTM

File: mkdocs.yml (hunk 5 above)
Comment (frbattid) [inline mkdocs.yml]: You are right, fixed in https://github.com/telefonicaid/fiware-cygnus/pull/949/commits/540563c64137c0bf10bcebaa9cf8f4dfe26093e1

File: doc/contributing/contributing_guidelines.md (hunk 4 above)
Comment (frbattid) [inline doc/contributing/contributing_guidelines.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/949/commits/540563c64137c0bf10bcebaa9cf8f4dfe26093e1

File: doc/contributing/contributing_guidelines.md (hunk 3 above)
Comment (frbattid) [inline doc/contributing/contributing_guidelines.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/949/commits/540563c64137c0bf10bcebaa9cf8f4dfe26093e1

File: doc/contributing/contributing_guidelines.md (hunk 2 above)
Comment (frbattid) [inline doc/contributing/contributing_guidelines.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/949/commits/540563c64137c0bf10bcebaa9cf8f4dfe26093e1

File: doc/cygnus-twitter/to_be_done.md (hunk 1 above)
Comment (frbattid) [inline doc/cygnus-twitter/to_be_done.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/949/commits/540563c64137c0bf10bcebaa9cf8f4dfe26093e1

**Comment by jmcanterafonseca** [comment]
LGTM

**Pull request merged**